### PR TITLE
Metrics: capture reorg distances and depths

### PR DESCRIPTION
**Pull request by terencechain**
**What does this PR do? Why is it needed?**
This PR captures reorg depth and reorg distance metrics. We'll like to monitor reorg and distance and depths using metrics. 
**Distance** is say you reorg x to y, how many steps it took to get from x to y. **Depth** is say how many steps it takes to get to a common ancestor

**Which issues(s) does this PR fix?**
N/A

**Other notes for review**
Gauge wasn't picked because of the inaccurate count. It depends on the sample rate of Prometheus. If Prometheus is down for a minute and two reorgs happen, we'll likely not be able to capture them all. Histogram seems like the best option

### CODE REVIEW

**Comment by prestonvanloon** [comment]
Is a gauge an appropriate choice for reorgs? Shouldn't it be a histogram?

**Comment by prestonvanloon** [comment]
How do you plan to use these metrics?

**Comment by terencechain** [comment]
> How do you plan to use these metrics?

I think gauge is more appropriate than histogram. I want to use it like this:
```
reorg_distance{component="beacon-chain",kubernetes_namespace=~"${namespace:pipe}", cluster=~"${cluster:pipe}", version=~"${version:pipe}"}
```

Similar to attestations in pool:

Screen Shot 2022-09-12 at 2 52 34 PM

File: beacon-chain/blockchain/metrics.go [hunk 1]
```diff
@@ -199,6 +199,20 @@ var (
...
+		prometheus.HistogramOpts{
+			Name:    "reorg_distance",
+			Help:    "Captures distance of reorgs",
+			Buckets: []float64{1, 2, 4, 8, 16, 32, 64},
+		},
+	)
+	reorgDepth = promauto.NewHistogram(
+		prometheus.HistogramOpts{
+			Name:    "reorg_depth",
+			Help:    "Captures depth of reorgs",
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/metrics.go:212]: Could you please add some details to the help text? I.e. what does depth / distance mean?

Comment (terencechain) [reply beacon-chain/blockchain/metrics.go]: > Could you please add some details to the help text? I.e. what does depth / distance mean?

Yep. Done

**Review by rauljordan — APPROVED** [review]

**Pull request merged**